### DISCUSSION TITLE
Confusion about `bind_to_cwd` and `cwd_target`

### DISCUSSION THREAD

**Discussion by wookayin** [nvim-neo-tree/neo-tree.nvim#1248] [Q&A]
Hi, I have some questions about automatic cwd management. I'm currently using the default config

```lua
    require("neo-tree").setup({
      filesystem = {
        bind_to_cwd = true, -- true creates a 2-way binding between vim's cwd and neo-tree's root
        cwd_target = {
          sidebar = "tab",   -- sidebar is when position = left or right
          current = "window" -- current is when position = current
        },
      }
    })
```

(1) I'd want 1-way binding between cwd and neotree root. I.e., I want neotree root automatically changes when `DirectoryChanged` event occurs, but not in the other way; i.e. do not change cwd when navigating the root in neotree filesystem.

How can I achieve this?

(2) If (1) is not possible, I'd like to change the CWD (as we move the root in the neo-tree explorer) **only for** the current window (`:lcd`), but only when the current position of the **neo-tree is "floating"**. `cwd_target` seems to be a relevant option for this, but this allows `sidebar` and `current` only; we do not have `floating` and floating is treated as if it were "sidebar" (left or right).

The current behavior can be quite annoying in the following scenario:

```vim
:cd ~/.dotfiles

" Open some file that is outside the current project/cwd
:vs $VIMRUNTIME/lua/vim/lsp/util.lua

" Open neotree in a floating window
:Neotree float reveal_force_cwd

" Close the pop-up window
:q

" Oops, now the CWD for tab has changed to $RUNTIME/lua/vim/lsp.
```

**Comment by cseickel** [comment]
Maybe you can achieve what you want by setting `bind_to_cwd = false` but then always open it with a `dir=` option to explicitly set the root to the current working directory.

**Reply by wookayin** [reply]
That would work, but I still would like to use the `:Neotree reveal` or similar commands. Looking forward to a new feature if it's yet to be implemented.
